### PR TITLE
Update podman_image to specify CPU arch when pulling image

### LINKED ISSUE #563: Allow setting the architecture of the image when pulling images
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind feature

**Description**

Allow setting the `--arch` flag when pulling images.

### PR DESCRIPTION
**Pull request by nishipy**
Fixes #563.

### CODE REVIEW

**Review by sshnaidm — COMMENTED** [review]
Thanks, just one nit. Awesome work on tests!

File: plugins/modules/podman_image.py [hunk 1]
```diff
@@ -526,9 +536,16 @@ def find_image(self, image_name=None):
...
+            inspect_json = self.inspect_image(image_name)
+            if self._is_target_arch(inspect_json, self.arch) or not self.arch:
+                return images
+
+        return None
+
+    def _is_target_arch(self, inspect_json=None, arch=None):
+        if arch and inspect_json[0]['Architecture'] == arch:
+            return True
+        return False
```
[the remark this answers is not in the document]
Comment (nishipy) [reply plugins/modules/podman_image.py:548]: Thanks!

**Comment by nishipy** [comment]
Oops, I have to add sign-off

**Comment by sshnaidm** [comment]
Amazing, thanks for contribution!

**Pull request merged**